fvm/executionParameters.go

```diff
@@ -33,17 +53,14 @@ func getMeterParameters(
 	meter.MeterParameters,
 	error,
 ) {
-	procParams := meter.DefaultParameters().
-		WithComputationLimit(uint(proc.ComputationLimit(ctx))).
-		WithMemoryLimit(proc.MemoryLimit(ctx)).
-		WithEventEmitByteLimit(ctx.EventCollectionByteSizeLimit)
+	procParams := getBasicMeterParameters(ctx, proc)
 
 	txnState := state.NewTransactionState(
 		view,
 		state.DefaultParameters().
 			WithMaxKeySizeAllowed(ctx.MaxStateKeySize).
 			WithMaxValueSizeAllowed(ctx.MaxStateValueSize).
-			WithMaxInteractionSizeAllowed(ctx.MaxStateInteractionSize))
+			WithMeterParameters(procParams))
```

Comment:
I'm not familiar with the different between `procParams` and `overrides`. 

But we are passing `procParams` here, and later updates the procParams if overrides.ComputationWeights is not nil (L74). However txnState won't see those updates, is that what we wanted? If so, why?

Reply:
The meter computation/memory weights (and memory limit) are dynamically configurable via these overrides.  (I'll give a more detailed answer about this in #3576, once I'm done responding to this PR)

Right now, we're using two different txnState in a single transaction:
1. The first txnState, without environment overrides, is used to fetch the environment overrides; as you pointed out, this txnState does not get updated.
2. The second txnState (created in fvm.go), with the environment overrides applied, is used to execute the actual transaction: `auth verification`, `seq num check`, and `body execution`.  Note that `get meter parameters` is not currently part of the transaction executor.  Also note that only `body execution` needs the updated meter parameters, the rest (`auth verification` and `seq num check`) don't need these overrides.

I'm planning to merge to the two txnState into one (and make `get meter parameters` part of the transaction executor).  `auth verification`, `seq num check`, and `get meter parameters` will use the basic meter parameters, and only the `body execution` will use the "enhanced" meter parameters.  See #2922 (and #3576) for additional discussion/detail.